elastalert/ruletypes.py

```diff
@@ -377,7 +377,12 @@ def handle_event(self, event, count, qk='all'):
             self.ref_window_filled_once = True
 
         if self.find_matches(self.ref_windows[qk].count(), self.cur_windows[qk].count()):
-            match = self.cur_windows[qk].data[-1][0]
+            # If placeholder (with 0 count) triggered alert, use a real event (the first one) instead of the latest (placeholder) event
+            if count == 0:
+                match = self.cur_windows[qk].data[0][0]
```

Comment:
I think technically data[0] could also be a placeholder event.
Perhaps better method might be something like
```
for match, count in self.cur_windows[qk].data:
    if count:
        break
```